src/fstab-generator/fstab-generator.c

```diff
@@ -1164,4 +1230,18 @@ static int run(const char *dest, const char *dest_early, const char *dest_late)
         return r < 0 ? r : r2 < 0 ? r2 : r3;
 }
 
-DEFINE_MAIN_GENERATOR_FUNCTION(run);
+static int run(int argc, char **argv) {
+        arg_sysroot_only = streq_ptr(argv[1], "--sysroot-only");
```

Comment:
I think it is better to use an environment variable, instead of the positional argument.

Reply:
I disagree. Environment variables are inherited and are useful for "global" settings and other defaults. A command-line argument is something that is clearly visible and is what should be used for something that completely changes the behaviour of a program.

(In fact, I'd say we rely on environment variables in systemd too much. E.g. their use a the sole way to enable debugging is imo hard to justify. But that's a topic for another time.)

Reply:
OK. That makes sense for me.